contracts/CreditGuild.sol

```diff
@@ -0,0 +1,105 @@
+//SPDX-License-Identifier: Unlicense
+pragma solidity ^0.8.4;
+
+import "hardhat/console.sol";
+import "@openzeppelin/contracts/token/ERC721/ERC721.sol";
+import "@openzeppelin/contracts/token/ERC20/ERC20.sol";
+import "@unioncredit/v1-sdk/contracts/BaseUnionMember.sol";
+
+/*
+credit guild
+*/
+contract CreditGuild is ERC721, BaseUnionMember { 
```

Comment:
The error here is telling me to mark the contract as abstract. I think it is because I haven't implemented the constructor in BaseUnionMember. This is the constructor for BaseUnionMember ```    constructor(address _marketRegistry, address _unionToken, address _underlyingToken) ```

Reply:
Yes it's likely that you constructor needs to implement the BaseUnionMember vars. Eg:

```solidity
  constructor (
    address member1, 
    address member2, 
    address member3,
    uint256 newMemberFee,
    address marketRegistry, 
    address unionToken, 
    address underlyingToken
    ) ERC721("UnionNFT", "UNFT") BaseUnionMember(marketRegistry, unionToken, underlyingToken) {
```